spec/fixtures/router_path_handling_tests.lua

```diff
@@ -0,0 +1,111 @@
+-- The following tests are used by unit and integration tests
+-- to test the router path handling. Putting them here avoids
+-- copy-pasting them in several places.
+--
+-- The tests can obtain this table by requiring
+-- "spec.fixtures.router_path_handling_tests"
+--
+-- Note that the tests are parsed into a hash form at the end
+-- of this file before they are returned
+
+local tests = {
+  -- upstream_path   route_path   strip_path  request_path    expected_path
```

Comment:
In general I think this is easier to read if it is:
```
-- request_path     route_path     service_path     expected_path     strip_path
```

`strip_path` can be on either side I think.